### DISCUSSION TITLE
How to use Qodana together with Pull Requests?

### DISCUSSION THREAD

**Discussion by shyim** [JetBrains/Qodana#192] [Q&A]
Hey,

how can I use Qodana with pull requests? When I don't pass a token, it does not start because of licensing. But I also don't want to provide the token to random external people. What is the process to use?

Example job: https://github.com/FriendsOfShopware/shopware-cli/actions/runs/5927979529/job/16072629482

**Comment by JustArchi** [comment]
Bump, I have the same problem. Is there any solution for this?

**Comment by SherkeyXD** [comment]
See https://youtrack.jetbrains.com/issue/QD-7310/Qodana-Scan-Always-Failed-on-Pull-Request-with-GitHub-Actions

The easiest way is to use eap versions, which doesn't require a token.

Unfortunately, `2023.2-eap` has just reached its end. We have to wait for the next eap version.

**Reply by shyim** [reply]
I have a license, but PRs by externals won't have my CI secrets. I think JetBrains made there a design failure in their licensing.

**Comment by mridang** [comment]
This is turning in to a major headache.
